### PR TITLE
Updates to Backpressure per initial PR comments

### PR DESCRIPTION
**Pull request by etspaceman**
CC @rossabaker @vasilmkd

### CODE REVIEW

File: std/shared/src/main/scala/cats/effect/std/Backpressure.scala [hunk 1]
```diff
@@ -56,29 +56,33 @@ object Backpressure {
       strategy: Strategy,
       bound: Int
   )(implicit GC: GenConcurrent[F, Throwable]): F[Backpressure[F]] = {
```
Comment (vasilmkd) [inline std/shared/src/main/scala/cats/effect/std/Backpressure.scala]: Is there a possibility to relax the `GenConcurrent` constraint to be `GenConcurrent[F, _]` like the other types in `std`? That does mean that you would have to rever the `require` change.

**Review by vasilmkd — COMMENTED** [review]
The diff is terrible for this change. 😆

**Comment by vasilmkd** [comment]
I take issue with the fact that the previous pull request was merged without explicit approval by anyone. I feel like we collectively dropped the ball there.

File: std/shared/src/main/scala/cats/effect/std/Backpressure.scala (hunk 1 above)
Comment (etspaceman) [reply std/shared/src/main/scala/cats/effect/std/Backpressure.scala]: Done. `MonadCancellable[F, Throwable]` is a requirement of `Resource.use`. So I replaced the `sem.permit` call with a bracket around `sem.acquire`

**Comment by djspiewak** [comment]
> I take issue with the fact that the previous pull request was merged without explicit approval by anyone. I feel like we collectively dropped the ball there.

We did, but that's mostly on me. I was kind of tacitly waiting for more feedback and @kubukoz broke the deadlock. We shouldn't make a habit of that kind of thing, but it wasn't the first step to go wrong.

Taking a step back to what I was pondering… my concern is that people will see this mechanism and think it's all they really need for backpressure, ignoring other higher-level tools like `Queue`. This is also a very general name (`Backpressure`) tailored at a very specific case (fiber supervision), which gives me a bit of pause. No one else seems to have strong opinions though so I could be very much over-indexing. :-)

**Comment by vasilmkd** [comment]
We live and we learn. As previously discussed, the train originally left the station when we shipped this interface in CE2. I personally was not in favor of it, even for CE2, but I didn't voice my opinion more loudly, so I guess that's ultimately fair.

**Review by vasilmkd — APPROVED** [review]

**Comment by kubukoz** [comment]
> > I take issue with the fact that the previous pull request was merged without explicit approval by anyone. I feel like we collectively dropped the ball there.
> 
> We did, but that's mostly on me. I was kind of tacitly waiting for more feedback and @kubukoz broke the deadlock. We shouldn't make a habit of that kind of thing, but it wasn't the first step to go wrong.

Sorry about that. It's really on me that I didn't check if every comment was resolved.

Can we enable required approvals in this repo? That should at least serve as a good hint that something is missing if there are no approvals.

**Comment by djspiewak** [comment]
> Can we enable required approvals in this repo? That should at least serve as a good hint that something is missing if there are no approvals.

Done. I generally don't like that restriction because it's annoying. :-P But you're right that it would have caught this problem.

**Pull request merged**